client/src/components/related_components/ProductCard.jsx

```diff
@@ -44,17 +70,31 @@ const ProductCard = (props) => {
   };
 
   const getRating = (ratings) => {
-    // console.log('ratings in getrating', ratings);
-    // cocounter = 0;
-    // var total = 0;
-    console.log(Object.entries(ratings));
+    let counter = 0;
+    let total = 0;
+
+    Object.entries(ratings).forEach((set) => {
+      const star = Number(set[0]);
+      const quantity = Number(set[1]);
+      total += star * quantity;
+      counter += quantity;
+    });
+
+    const avg = total / counter;
+    const round = (Math.round(avg * 4) / 4).toFixed(2);
+    setRating(round);
   };
```

Comment:
We should try to re-factor our calcRating function in App.js so that we can pass it around to components like here, since the logics the same.